backend/src/rtp.rs

```diff
@@ -1775,29 +1837,34 @@ impl Endpoint {
     ) -> Option<Packet<Vec<u8>>> {
         let tcc_sender = &mut self.tcc_sender;
         let rtx_sender = &mut self.rtx_sender;
-        let rtx = rtx_sender.resend_as_rtx(ssrc, seqnum, || tcc_sender.increment_seqnum())?;
-        self.encrypt_and_send_rtp(rtx, now)
+        let rtx = rtx_sender
+            .resend_as_rtx(ssrc, seqnum, || tcc_sender.increment_seqnum())
+            .ok_or_else(|| EndpointError::UndeterminedError("Packet Error".to_string()))
+            .ok()?;
+        self.encrypt_and_send_rtp(rtx, now).ok()
     }
 
     pub fn send_padding(&mut self, rtx_ssrc: Ssrc, now: Instant) -> Option<Packet<Vec<u8>>> {
         let tcc_seqnum = self.tcc_sender.increment_seqnum();
         let padding = self.rtx_sender.send_padding(rtx_ssrc, tcc_seqnum);
-        self.encrypt_and_send_rtp(padding, now)
+        self.encrypt_and_send_rtp(padding, now).ok()
     }
 
     fn encrypt_and_send_rtp(
         &mut self,
         mut outgoing: Packet<Vec<u8>>,
         now: Instant,
-    ) -> Option<Packet<Vec<u8>>> {
+    ) -> Result<Packet<Vec<u8>>, EndpointError> {
         // Remember the packet sent for RTX before we encrypt it.
         if is_rtxable_payload_type(outgoing.payload_type()) {
             self.rtx_sender.remember_sent(outgoing.to_owned(), now);
         }
         // Don't remember the packet sent for TCC until after we actually send it.
         // (see remember_sent_for_tcc)
-        outgoing.encrypt_in_place(&self.encrypt.rtp.key, &self.encrypt.rtp.salt)?;
-        Some(outgoing)
+        outgoing
+            .encrypt_in_place(&self.encrypt.rtp.key, &self.encrypt.rtp.salt)
+            .map_err(|e| EndpointError::UndeterminedError(e.to_string()))?;
```

Comment:
Did you mean to use `SrtpEncryptionError` here?

Reply:
wasn't very sure about it so left it under that case, for further feedback.